packages/@sanity/cli/src/util/remoteTemplate.ts

```diff
@@ -0,0 +1,508 @@
+ * - At least one package must have sanity.config.js/ts and sanity.cli.js/ts
+ * - Each package must have a .env.template, .env.example, or .env.local.example
+ *
+ * For single-package repositories:
+ * - Must have a valid package.json with 'sanity' dependency
+ * - Must have sanity.config.js/ts and sanity.cli.js/ts
+ * - Must have .env.template, .env.example, or .env.local.example
+ *
+ * Environment files must include:
+ * - SANITY_PROJECT_ID or SANITY_STUDIO_PROJECT_ID variable
+ * - SANITY_DATASET or SANITY_STUDIO_DATASET variable
+ *
+ * @throws Error if validation fails with specific reason
+ */
+export async function validateRemoteTemplate(
+  repoInfo: RepoInfo,
+  packages: string[] = [''],
+  bearerToken?: string,
+): Promise<void> {
+  const {username, name, branch, filePath} = repoInfo
+  const baseUrl = `https://raw.githubusercontent.com/${username}/${name}/${branch}/${filePath}`
+
+  const headers: Record<string, string> = {}
+  if (bearerToken) {
+    headers.Authorization = `Bearer ${bearerToken}`
+  }
+
+  const validations = await Promise.all(
+    packages.map((pkg) => validatePackage(baseUrl, pkg, headers, pkg === '')),
+  )
+
+  const hasSanityDep = validations.some((v) => v.hasSanityDep)
+  if (!hasSanityDep) {
+    throw new Error('At least one package must include "sanity" as a dependency in package.json')
+  }
+
+  const hasSanityConfig = validations.some((v) => v.hasSanityConfig)
+  if (!hasSanityConfig) {
+    throw new Error('At least one package must include a sanity.config.js or sanity.config.ts file')
+  }
+
+  const hasSanityCli = validations.some((v) => v.hasSanityCli)
+  if (!hasSanityCli) {
+    throw new Error('At least one package must include a sanity.cli.js or sanity.cli.ts file')
+  }
+
+  const missingEnvPackages = packages.filter((pkg, i) => !validations[i].hasEnvFile)
+  if (missingEnvPackages.length > 0) {
+    throw new Error(
+      `The following packages are missing .env.template, .env.example, or .env.local.example files: ${missingEnvPackages.join(
+        ', ',
+      )}`,
+    )
+  }
+}
+
+export async function isNextJsTemplate(root: string): Promise<boolean> {
+  try {
+    const packageJson = await readFile(join(root, 'package.json'), 'utf8')
+    const pkg = JSON.parse(packageJson)
+    return !!(pkg.dependencies?.next || pkg.devDependencies?.next)
+  } catch {
+    return false
+  }
+}
+
+export async function applyEnvVariables(
+  root: string,
+  envData: EnvData,
+  targetName = '.env',
+): Promise<void> {
+  const templatePath = await Promise.any(
+    ENV_TEMPLATE_FILES.map(async (file) => {
+      await access(join(root, file))
+      return file
+    }),
+  ).catch(() => {
+    throw new Error('Could not find .env.template, .env.example or .env.local.example file')
+  })
+
+  try {
+    const templateContent = await readFile(join(root, templatePath), 'utf8')
+    const {projectId, dataset, readToken = '', apiVersion = 'vX'} = envData
```

Comment:
We should default to current `YYYY-MM-DD`, not `vX` no?

Reply:
I'm actually leaning towards the template setting this, since the template it self might relying on certain api features

Reply:
Removed it altogether

Reply:
Ah, yeah thats a great point, should def be the responsibility of the template.